spec/acceptance/suites/offline/00_default_spec.rb

```diff
@@ -0,0 +1,158 @@
+  hosts.each do |host|
+    context "on #{host}" do
+      let(:vagrant_rpm) { 'https://releases.hashicorp.com/vagrant/2.2.3/vagrant_2.2.3_x86_64.rpm' }
+      let(:build_user) { 'build_user' }
+      let(:build_user_cmd) { "runuser #{build_user} -l -c" }
+
+      # Not sure if this is a QEMU thing with the image or something else
+      it 'works around a CentOS curl bug with libvirt' do
+        on(host, %(touch /etc/sysconfig/64bit_strstr_via_64bit_strstr_sse2_unaligned))
+      end
+
+      it 'adds the build user' do
+        on(host, %(useradd -b /home -G wheel -m -c "Build User" -s /bin/bash -U #{build_user}))
+
+        # Allow the build user to perform privileged operations
+        on(host, %(echo 'Defaults:build_user !requiretty' >> /etc/sudoers))
+      end
+
+      it 'installs required packages' do
+        required_packages = [
+          'augeas-devel',
+          'autoconf',
+          'automake',
+          'bison',
+          'createrepo',
+          'curl',
+          'gcc',
+          'gcc-c++',
+          'genisoimage',
+          'git',
+          'glibc-devel',
+          'glibc-headers',
+          'gnupg2',
+          'libffi-devel',
+          'libicu-devel',
+          'libtool',
+          'libvirt',
+          'libvirt-client',
+          'libvirt-devel',
+          'libxml2',
+          'libxml2-devel',
+          'libxslt',
+          'libxslt-devel',
+          'libyaml-devel',
+          'ntpdate',
+          'openssl',
+          'openssl-devel',
+          'qemu',
+          'readline-devel',
+          'rpm-build',
+          'rpm-sign',
+          'rpmdevtools',
+          'ruby-devel',
+          'rubygems',
+          'seabios',
+          'sqlite-devel',
+          'util-linux',
+          'which'
+        ]
+
+        host.install_package('epel-release')
+
+        on(host, %(yum -y install #{required_packages.join(' ')}))
+        on(host, %(yum -y update))
+      end
+
+      it 'removes limits from the system' do
+        # Remove system limits
+        on(host, %(rm -rf /etc/security/limits.d/*.conf))
+      end
+
+      it 'start libvirt' do
+        on(host, %(puppet resource service libvirtd ensure=running enable=true))
+      end
+
+      it 'adds the build user to the libvirt groups' do
+        on(host, %(usermod -a -G qemu -G libvirt #{build_user}))
+      end
+
+      it 'installs RPM for the build user' do
+        # Install RVM
+        on(host, %(#{build_user_cmd} "for i in {1..5}; do { gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3; } && { gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB; } && break || sleep 1; done"))
```

Comment:
_(Speaking as one who has been responsible for much of this gpg-keyserver-roulette-hell-logic):_ let's rip all this out and just ship the GPG keys we trust along with the project!

Extra credit: Throw in an ENV var to change the default key and rvm-installer, and we have a hook for running (this part) of the tests in a network-isolated environment.